doc/manual/manual.tex

```diff
@@ -1595,14 +1741,198 @@ \subsubsection{Compressible models}
+
+\begin{itemize}
+
+\item
+``incompressible'':
+\[
+ \nabla \cdot \textbf{u} = 0,
+\]
+
+\item
+``isothermal compression'':
+\[
+ \nabla \cdot \textbf{u} = -\frac{1}{\rho} \frac{\partial \rho}{\partial p} \rho \textbf{g} \cdot \textbf{u},
+\]
+where the compressibility $\kappa = \frac{1}{\rho} \frac{\partial \rho}{\partial p}$ 
+is defined in the material model. 
+
+\item
+``reference density profile'':
+\[
+ \nabla \cdot \textbf{u} = -\frac{1}{\bar{\rho}} \frac{\partial \bar{\rho}}{\partial z} \frac{\textbf{g}}{\|\textbf{g}\|} \cdot \textbf{u},
+\]
+where reference profiles for the density $\bar{\rho}$ and the density gradient $\frac{\partial \bar{\rho}}{\partial z}$
+provided by \text{AdiabaticConditions}
+are used. Note that the
+gravity is assumed to point downwards in depth direction
+
+\item
+``implicit reference density profile'':
+\[
+ \nabla \cdot \textbf{u} + \frac{1}{\bar{\rho}} \frac{\partial \bar{\rho}}{\partial z} \frac{\textbf{g}}{g} \cdot \textbf{u} = 0,
+\]
+which uses the same approximation for the density as ``reference density profile'', 
+but implements this term on the left-hand side instead of the right-hand side of the mass
+conservation equation. This effectively uses the current velocity $\textbf{u}$ instead
+of an extrapolated velocity from the last timesteps.
+
+ \item 
+``ask material model'', which uses ``isothermal compression'' if the material model is compressible 
+(as reported by the \texttt{is\_compressible()} function) and ``incompressible'' otherwise.
+\end{itemize}
+
+
+\paragraph*{On the stress tensor approximation.}
+
+Incompressibility in the mass conservation equation automatically simplifies the shear strain rate in the momentum and temperature equation from
+\[
+ \nabla \cdot \tau = 
+ \nabla \cdot \left[
+ 2\eta \left(\varepsilon(\mathbf u)
+                                  - \frac{1}{3}(\nabla \cdot \mathbf u)\mathbf 1\right)
+                                  \right]
+\]
+to
+\[
+  \nabla \cdot \tau =  \nabla \cdot \left[
+ 2\eta \varepsilon(\mathbf u) \right].
+\]
+
+\subsubsection{Temperature equation approximation}
+
+The density occurs multiple times in the temperature equation. Depending on the selected approximation it is computed in two different ways:
+
+\begin{itemize}
+ \item 
+``real density'': Use the full density as computed for the buoyancy term of the momentum equation (i.e. the return value of the material model for ``density``),
+
+\item
+``reference density profile'': Use the density as computed for the reference profile (which can be constant, an adiabatic profile, or an entirely different function, and is determined by the adiabatic conditions model).
+\end{itemize}
+
+\subsubsection{Approximation of the buoyancy term}
+The buoyancy term (right-hand side of the momentum equation) always uses the 
+density that is provided by the material model (see Section~\ref{sec:material-models}). 
+There, one can decide to use a density that for example depends on temperature 
```

Comment:
"There" -- where? You mean in the material model?

It's also not quite clear what "one can decide" means. I suppose you mean that one can decide this when implementing a new material model?